src/Adapter/Http/Repository/HttpPodRepository.php

```diff
@@ -35,13 +38,23 @@ class HttpPodRepository implements PodRepository
     private $namespaceClient;
 
     /**
-     * @param HttpConnector       $connector
+     * @var PodAwareStatusProvider
+     */
+    private $statusProvider;
+
+    /**
+     * @param HttpConnector $connector
      * @param HttpNamespaceClient $namespaceClient
+     * @param PodStatusProviderFactory $statusProviderFactory
      */
-    public function __construct(HttpConnector $connector, HttpNamespaceClient $namespaceClient)
-    {
+    public function __construct(
+        HttpConnector $connector,
+        HttpNamespaceClient $namespaceClient,
+        PodStatusProviderFactory $statusProviderFactory
+    ) {
         $this->connector = $connector;
         $this->namespaceClient = $namespaceClient;
+        $this->statusProvider = $statusProviderFactory->createFromRepository($this);
```

Comment:
Not sure of the value of the `StatusProviderFactory` here, if it's just to do a `new [...]` :)

Reply:
Well, this is a way to break the circular reference: `PodRepository` needs `PodStatusProvider` and vice versa. Introducing a factory class eliminates the circular dependency.

Reply:
~eliminates~ hides 😉

Reply:
Call it whatever you like. What I can do about it? :)

Reply:
@sroze My proposal is to move the `HttpPodRepository::attach()` and `HttpPodRepository::streamOutput()` methods to the `PodStatusProvider`, because these methods does not really belong to a repository. Then `HttpPodRepository` does not need to have `PodStatusProvider` any more.
But that would be a backward compatibility breaking change, so the `attach()` method should stay for a while (and it would result duplicated code until removed).
What do you think?